lib/auth/auth_with_roles.go

```diff
@@ -5600,7 +5600,28 @@ func (a *ServerWithRoles) CreateSAMLIdPServiceProvider(ctx context.Context, sp t
 		return trace.Wrap(err)
 	}
 
-	return a.authServer.CreateSAMLIdPServiceProvider(ctx, sp)
+	err := a.authServer.CreateSAMLIdPServiceProvider(ctx, sp)
+	if err != nil {
+		return trace.Wrap(err)
```

Comment:
Are you intentionally not emitting an event if the create failed? 
Feels like we'd probably want an event in that case too.

(Same goes for the update and delete calls below)

Reply:
Not intentional -- sounds good, I'll add a success flag to this.